packages/editor/src/components/post-url/index.js

```diff
@@ -0,0 +1,108 @@
+/**
+ * WordPress dependencies
+ */
+import { useSelect, useDispatch } from '@wordpress/data';
+import { safeDecodeURIComponent, cleanForSlug } from '@wordpress/url';
+import { useState } from '@wordpress/element';
+import { __experimentalInspectorPopoverHeader as InspectorPopoverHeader } from '@wordpress/block-editor';
+import { __ } from '@wordpress/i18n';
+import { TextControl, ExternalLink } from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import { store as editorStore } from '../../store';
+
+export default function PostURL( { onClose } ) {
+	const { isEditable, postSlug, postLink, permalinkPrefix, permalinkSuffix } =
+		useSelect( ( select ) => {
+			const permalinkParts = select( editorStore ).getPermalinkParts();
+			return {
+				isEditable: select( editorStore ).isPermalinkEditable(),
+				postSlug: safeDecodeURIComponent(
+					select( editorStore ).getEditedPostSlug()
+				),
+				postLink: select( editorStore ).getCurrentPost().link,
```

Comment:
Suggestion (non-blocking): I know it's all about personal preference, but why not extract selectors at the top and avoid repeating `select( editorStore )`.

Reply:
Right now I'm in a _don't destructure_ phase. I alternate constantly 😂

Advantages of destructuring:

- Less repetition of the `select()`.

Advantages of not destructuring:

- More explicit that it's a selector and not a function.
- When you add an early `return`, you don't run into annoying "you must declare this immediately before first use" lint errors which sucks cause then all the destructuring isn't grouped together.

Reply:
Good points! I'm counting this one as resolved 😄

Reply:
I'll change my preference in a month you watch.